#This is how orthogroups were categorized for Woodruff et al. "Widespread changes in gene expression accompany body size evolution in nematodes"

#This is the second part, the first part is "orthologous_genes_notes.sh"

#set working directory
setwd("/Users/gavin/genome/transcriptome_revisions_12-2023/orthologs")

#get orthogroups+Gene ID data into R
    #generated by OrthoFinder and orthologous_genes_notes.sh . this file is here: https://github.com/gcwoodruff/inopinata_developmental_transcriptomics_2023/blob/main/G3_revisions_1/orthologs/orthogroups_genes_w_orphans_sort_uniq.tsv

eleg_inop_OGs <- read.table("orthogroups_genes_w_orphans_sort_uniq.tsv",header=FALSE,sep='\t')

#Get orthogroup category data into R.
    #generated by orthologous_genes_notes.sh. this file is here: https://github.com/gcwoodruff/inopinata_developmental_transcriptomics_2023/blob/main/G3_revisions_1/orthologs/orthogroup_categories_sort_uniq.tsv
categories_OGs <- read.table("orthogroup_categories_sort_uniq.tsv",header=FALSE,sep='\t')

#JOIN THE FILES
joined_OGs <- merge(eleg_inop_OGs,categories_OGs,by.x="V1",by.y="V1")

#get inopinata OG id, OG catgory, and OG gene ids!!!
inopinata <- data.frame(OG = joined_OGs$V1, Category=joined_OGs$V2.y,Genes=joined_OGs$V4)
#get elegans OG id, OG catgory, and OG gene ids!!!
elegans <- data.frame(OG = joined_OGs$V1, Category=joined_OGs$V2.y,Genes=joined_OGs$V3)

#these data are used for other things
write.table(inopinata,"inopinata_ogs_categories.tsv",sep='\t', quote=FALSE,row.names = FALSE, col.names = FALSE)
write.table(elegans,"elegans_ogs_categories.tsv",sep='\t', quote=FALSE,row.names = FALSE, col.names = FALSE)

#alright, so, all the genes are bunched together on a single row.

#can we make a file where each gene gets its own row? this would be more helpful

#make a null variable for the loop. we'll add data to it in the loop
big_inopinata_df <- NULL
#just get orthogroups where there are genes for inopinata
inopinata_genes_present <- inopinata[inopinata$Genes != "", ]

    #for all inopinata rows do
for(i in 1:nrow(inopinata_genes_present)) {
    #get a row
    row <- inopinata_genes_present[i,]
    #split that row into more rows wherever there is a comma in the genes column
    row_genes <- as.vector(strsplit(row$Genes,", "))
    #make a new data frame: 
        #gene id
        #OG id
        #OG category
    row_df <- data.frame(Gene=row_genes[[1]],OG=rep(row$OG, length(row_genes[[1]])),Category=rep(row$Category, length(row_genes[[1]])))
    #add it to our big df and repeat
    big_inopinata_df <- rbind(big_inopinata_df,row_df)
}
#write that to a file
write.table(big_inopinata_df,"inopinata_GENES_ogs_categories.tsv",sep='\t', quote=FALSE,row.names = FALSE, col.names = FALSE)

#now elegans
#make a null variable for the loop. we'll add data to it in the loop
big_elegans_df <- NULL
#just get orthogroups where there are genes for elegans
elegans_genes_present <- elegans[elegans$Genes != "", ]
    #for all elegans rows do
for(i in 1:nrow(elegans_genes_present)) {
    #get a row
    row <- elegans_genes_present[i,]
    #split that row into more rows wherever there is a comma in the genes column
    row_genes <- as.vector(strsplit(row$Genes,", "))
    #make a new data frame: 
        #gene id
        #OG id
        #OG category
    row_df <- data.frame(Gene=row_genes[[1]],OG=rep(row$OG, length(row_genes[[1]])),Category=rep(row$Category, length(row_genes[[1]])))
    #add it to our big df and repeat
    big_elegans_df <- rbind(big_elegans_df,row_df)
}
#write that to a file
write.table(big_elegans_df,"elegans_GENES_ogs_categories.tsv",sep='\t', quote=FALSE,row.names = FALSE, col.names = FALSE)
